An existing file should not generate an error for all functions.
This patch removes check against FILE_EXIST where not valid.

Signed-off-by: Jonas Karlsson <jonas@gobolinux.org>

--- funionfs-0.4.3/main.c	2007-01-18 20:35:52.000000000 -0200
+++ funionfs-0.4.3-new/main.c	2008-08-09 21:42:36.000000000 -0300
@@ -1,4 +1,3 @@
-
 /*
     FUNIONFS: UNIONFS over FUSE Usermode filesystem
     Copyright (C) 2005-2006  Stephane APIOU <stephane.apiou@free.fr>
@@ -294,11 +293,6 @@
 				break;
 			}
 		}
-		else if (status & FILE_EXIST)
-		{
-			errno = EEXIST;
-			break;
-		}
 		else
 		{
 			res = funionfs_pathtowrite(path);
@@ -697,11 +691,6 @@
 
 	do
 	{
-		if (status & FILE_EXIST)
-		{
-			errno = EEXIST;
-			break;
-		}
 		if (!(status & FILE_CREATABLE))
 		{
 			errno = EROFS;
@@ -750,11 +739,6 @@
 
 	do
 	{
-		if (status & FILE_EXIST)
-		{
-			errno = EEXIST;
-			break;
-		}
 		if (!(status & FILE_CREATABLE))
 		{
 			errno = EROFS;
@@ -803,11 +787,6 @@
 
 	do
 	{
-		if (status & FILE_EXIST)
-		{
-			errno = EEXIST;
-			break;
-		}
 		if (!(status & FILE_CREATABLE))
 		{
 			errno = EROFS;
@@ -855,11 +834,6 @@
 
 	do
 	{
-		if (status & FILE_EXIST)
-		{
-			errno = EEXIST;
-			break;
-		}
 		if (!(status & FILE_CREATABLE))
 		{
 			errno = EROFS;
@@ -1319,11 +1293,6 @@
 
 	do
 	{
-		if (status & FILE_EXIST)
-		{
-			errno = EEXIST;
-			break;
-		}
 		if (!(status & FILE_CREATABLE))
 		{
 			errno = EROFS;
@@ -1442,11 +1411,6 @@
 
 	do
 	{
-		if (status & FILE_EXIST)
-		{
-			errno = EEXIST;
-			break;
-		}
 		if (!(status & FILE_CREATABLE))
 		{
 			errno = EROFS;
